Gems/AWSCore/Code/Include/Public/Framework/ServiceRequestJob.h

```diff
@@ -380,9 +380,9 @@ namespace AWSCore
                     responseContent.c_str()
                 );
 
-                // This is determined by AZ::g_maxMessageLength defined in in dev\Code\Framework\AzCore\AzCore\Debug\Trace.cpp.
-                // It has the value 4096, but there is the timestamp, etc., to account for so we reduce it by a few characters.
-                const int MAX_MESSAGE_LENGTH = 4096 - 128;
+                // This is determined by the SLogMsg struct defined in Code\Legacy\CrySystem\Log.h.
+                // The maximum size of a log message is 512 characters, but there is the timestamp, etc., to account for so we reduce it by a few characters.
+                const int MAX_MESSAGE_LENGTH = 512 - 128;
```

Comment:
1. instead of hardcoding the magic number, can we define global variable for the size in SLogMsg? so it can be referred here
or 2. there should be an util function inside log class to sanity check log message if there is size limitation, then other components can use it directly

we should have at least one of above methods. (unit test is not required, as we should not have responsibility to test log component in unit test)

Reply:
Sounds like two requests we should send to the team which owns the log system. Root cause is because of the following line in Code\Legacy\CrySystem\Log.cpp:

SLogMsg msg;
azstrcpy(msg.msg, AZ_ARRAY_SIZE(msg.msg), szString);

It triggers an assert when the size of szString exceeds AZ_ARRAY_SIZE(msg.msg) and from the log trace and azstrcpy is calling strcpy_s underneath.

There're some modifications in the logging system recently which causes the log size limit changed from AZ::g_maxMessageLength to 512 and seems like Esteban is still updating the legacy system. Will loop him in for suggestions.